src/planning.rs

```diff
@@ -202,6 +205,9 @@ impl<'fetcher> BuildPlannerImpl<'fetcher> {
         continue;
       }
 
+      let own_source_id = own_package.source.clone()
```

Comment:
Nit: I think you can replace `source.clone()` with `source.as_ref()` as you don't actually need a clone of the string to deserialize, just a clone of the option. `as_ref` will do that for you (it gives you a new Option<&T> that you can map/unwrap/whatever.